concept/type_/type_manager/type_cache/struct_definition_cache.rs

```diff
@@ -0,0 +1,37 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+use bytes::Bytes;
+use encoding::graph::definition::{definition_key::DefinitionKey, r#struct::StructDefinition, DefinitionValueEncoding};
+use storage::{key_range::KeyRange, snapshot::ReadableSnapshot};
+
+pub struct StructDefinitionCache {
```

Comment:
Right now we refer to a "cache" as generally a collection of many cache entries, not the individual entries this are cached. Perhaps we can find a different name for this? `StructDefinitionCacheEntry`? 

Then the Impl block doesn't make sense anymore, but you can have a standalone function that creates the cache (Box<Option<[Entry]>>)